Gauge.CSharp.Lib/CommonDataStore.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+
+namespace Gauge.CSharp.Lib
+{
+    public class CommonDataStore
```

Comment:
Rather than using this pattern of a static builder, may I suggest we consider using extension methods to remove the duplicates? Here's my crude attempt"

```
using System.Collections.Concurrent;
using System.Threading;

namespace Gauge.CSharp.Lib
{
    public static class DataStoreExtensions
    {
        internal static object Get(this ThreadLocal<ConcurrentDictionary<object, object>> store, string key)
        {
            lock (store)
            {
                object outVal;
                var valueExists = store.Value.TryGetValue(key, out outVal);
                return valueExists ? outVal : null;
            }
        }

        internal static T Get<T>(this ThreadLocal<ConcurrentDictionary<object, object>> store, string key)
        {
            lock (store)
            {
                return (T)store.Get(key);
            }
        }

        internal static void Add(this ThreadLocal<ConcurrentDictionary<object, object>> store, string key, object value)
        {
            lock (store)
            {
                store.Value[key] = value;

            }
        }

        internal static void Clear(this ThreadLocal<ConcurrentDictionary<object, object>> store)
        {
            lock (store)
            {
                store.Value.Clear();
            }
        }
    }
}
```

and the `ScenarioDataStore` would then look like:

```
using System.Collections.Concurrent;
using System.Threading;

namespace Gauge.CSharp.Lib
{
    public class ScenarioDataStore
    {
        private static ThreadLocal<ConcurrentDictionary<object, object>> store = new ThreadLocal<ConcurrentDictionary<object, object>>(() => new ConcurrentDictionary<object, object>());

        public static object Get(string key)
        {
            return store.Get(key);
        }

        public static T Get<T>(string key)
        {
            return store.Get<T>(key);
        }

        public static void Add(string key, object value)
        {
            store.Add(key, value);
        }

        public static void Clear()
        {
            store.Clear();
        }
    }
}
```

What do you think?

Reply:
Ah, true (I am not used to it yet). it is excellent. I will apply it. thanks million,